src/Roave/SecurityAdvisories/Version.php

```diff
@@ -37,85 +38,90 @@ public static function fromString(string $version) : self
+    }
 
-            return $thisVersion > $otherVersion;
-        }
+    public function isGreaterOrEqualThan(self $other) : bool
+    {
+        [$first, $second] = $this->normalizeVersions($other);
+        return version_compare($first, $second, '>=');
+    }
 
-        return (bool) array_filter(array_slice($this->versionNumbers, count($other->versionNumbers)));
+    public function toString()
+    {
+        return $this->version;
     }
 
     /**
-     * Compares two versions and sees if this one is greater or equal than the given one
-     *
-     * @todo may become a simple array comparison (if PHP supports it)
-     *
-     * @param Version $other
-     *
-     * @return bool
+     * Here we need to append zeroes so comparison will work correctly
      */
-    public function isGreaterOrEqualThan(self $other) : bool
+    private function normalizeVersions(Version $other): array
     {
-        foreach ($other->versionNumbers as $index => $otherVersion) {
-            $thisVersion = $this->versionNumbers[$index] ?? 0;
-
-            if ($thisVersion === $otherVersion) {
-                continue;
-            }
-
-            return $thisVersion > $otherVersion;
+        [$versionA, $tailA] = $this->stabilizeVersion($this->version);
+        [$versionB, $tailB] = $this->stabilizeVersion($other->version);
+
+        [$countA, $countB, $diff] = $this->getVersionStats($versionA, $versionB);
+        switch ($countA <=> $countB) {
+            case -1:
+                return [
+                    $versionA.str_repeat('.0', $diff).$tailA,
+                    $versionB.$tailB
+                ];
+            case 0;
+                return [
+                    $versionA.$tailA,
+                    $versionB.$tailB
+                ];
+            case 1;
+                return [
+                    $versionA.$tailA,
+                    $versionB.str_repeat('.0', $diff).$tailB
+                ];
         }
-
-        return true;
     }
 
-    public function getVersion() : string
+    /**
+     * Split version string representation in two parts - stable and stability tail,
+     * if no tail is present return null
+     */
+    private function stabilizeVersion(string $version) : array
     {
-        return implode('.', $this->versionNumbers);
+        $regExp = '/'.self::STABILITY_TAIL.'$/';
+        preg_match($regExp, $version, $matches, PREG_OFFSET_CAPTURE);
+
+        if (!is_null($matches[0][0])) {
+            return [substr($version, 0, $matches[0][1]), $matches[0][0]];
+        }
+
+        return [$version, null];
     }
 
     /**
-     * @param int[] $versionNumbers
-     *
-     * @return int[]
+     * Get count of version numbers for each version
+     * also return difference in version numbers between versions
      */
-    private static function removeTrailingZeroes(array $versionNumbers) : array
+    private function getVersionStats(string $versionA, string $versionB) : array
```

Comment:
The naming of this method is a bit misleading. It seems to return the number of version significant numbers, but doesn't include stability constraints in the count. Very much difficult to follow...

Reply:
yup, I'm terrible at naming and probably programming )
this function will return count of version numbers for both "versions" and also the difference between them

how about I delete `getVersionStats` and keep the logic in `normalizeVersions` ?
though function body will be more bloated :confused:

Reply:
What are "Stats" then? ;-)

Reply:
oh I have no idea, it was too late and I was not doing very well : )